docker/Dockerfile.psga-pipeline

```diff
@@ -1,4 +1,4 @@
-FROM condaforge/mambaforge:4.14.0-0 AS condabuild
+FROM condaforge/mambaforge:23.3.1-1 AS condabuild
```

Comment:
Do we need to pin to this exact version? Can we relax this to just the minor version so we get security updates for free?

Reply:
they don't seem to ship an image that gets security updates, I am unsure how docker would handle it (as these are image tags and not 'real' version numbers)